File: problems/python/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        next = None
        tmp = None
        while head:
            tmp = ListNode(head.val)
            if next:
                tmp.next = next
            next = tmp
            head = head.next
        return tmp

File: problems/cpp/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *next = NULL;
        ListNode *tmp = NULL;
        while(head){
            tmp = new ListNode(head->val);
            if(next){
                tmp->next = next;
            }
            next = tmp;
            head = head->next;
        }
        return tmp;

    }
};